### PR TITLE
[cmd/opampsupervisor]: Configure the `ppid` option in the opamp extension

### LINKED ISSUE #32189: [cmd/opampsupervisor] Supervisor may leak collector process
### Component(s)

cmd/opampsupervisor, extension/opamp

### Describe the issue you're reporting

In its current form, the Supervisor may leak a collector process if it is unexpectedly killed and doesn't get a chance to stop the collector. You can see this by issuing a `kill -9` to the supervisor and observing that the collector process is still running. This will block subsequent startups of the supervisor, as multiple collectors will try to occupy the same ports (8888, any user-configured components with a port)

Under normal circumstance, we shouldn't leak the collector process, but I think we can make this more robust so that when the supervisor unexpectedly dies, the collector dies too.

I thought through a couple ideas (collector PID file, doing some black magic stuff per-os to have the OS auto-kill the children), but I think they all end up a way more complex than just having the collector monitor it's ppid and exiting if it changes. 

In one of our old managed agents, this is how we would handle the cases where the supervisor may shutdown without properly killing the child process.

To fully outline the proposal, it would be something like:

1. Add an optional field to the OpAMP extension, `supervisor_pid`
2. When `supervisor_pid` is configured, the OpAMP extension will poll (maybe every ~5 seconds) and ensure that the value of `os.Getppid()` equals the value of `supervisor_pid`
3. If these values are not equal, the OpAMP extension reports a fatal error to trigger a collector shutdown.

Note: I think the second step here might need different logic on Windows, as the orphaned process may not be re-parented like it is on other systems.

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
**Description:** <Describe what has changed.>
* Configures the PPID of the opamp extension in the supervisor. This allows the collector to detect if the supervisor exits and shut itself down.

**Link to tracking Issue:** Closes #32189

**Testing:** <Describe what testing was performed and which tests were added.>
* Manually tested by starting the supervisor, then kill -9'ing the supervisor. The collector previously would have still been running, but now shuts itself down. Doing this you can also see the following log:
```
2024-05-06T14:52:31.010-0400	error	otelcol@v0.99.1-0.20240503221155-67d37183e6ac/collector.go:278	Asynchronous error received, terminating process	{"error": "collector was orphaned, process with pid 38908 does not exist"}
```

**Documentation:** <Describe the documentation added.>
Added `orphan_detection_interval` to the spec as a configurable option

### CODE REVIEW

File: cmd/opampsupervisor/specification/README.md [hunk 1]
```diff
@@ -123,7 +123,10 @@ collector:
   # Path to Collector executable. Required.
   executable: /opt/otelcol/bin/otelcol
 
-  # Extra command line flags to pass to the Collector executable.
+  # The interval on which the collector checks to see if it's been orphaned.
```
Comment (tiffany76) [inline cmd/opampsupervisor/specification/README.md:126]:
```suggestion
  # The interval on which the Collector checks to see if it's been orphaned.
```

File: cmd/opampsupervisor/specification/README.md [hunk 2]
```diff
@@ -123,7 +123,10 @@ collector:
   # Path to Collector executable. Required.
   executable: /opt/otelcol/bin/otelcol
 
-  # Extra command line flags to pass to the Collector executable.
+  # The interval on which the Collector checks to see if it's been orphaned.
+  orphan_detection_interval: 5s
+
+# Extra command line flags to pass to the Collector executable.
```
Comment (andrzej-stencel) [inline cmd/opampsupervisor/specification/README.md:129]:
```suggestion
  # Extra command line flags to pass to the Collector executable.
```

**Review by andrzej-stencel — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Comment by evan-bradley** [comment]
Could you resolve the conflicts?

**Comment by BinaryFissionGames** [comment]
@evan-bradley I've resolved the conflicts and merged in the latest changes from main.

**Pull request merged**